### DISCUSSION TITLE
Completely break inheritance (permission & sharing) and apply explicit permission to a single file in ODB

### DISCUSSION THREAD

**Discussion by msrosellinpo** [pnp/powershell#428] [Q&A]
Hi all,
I have a problem that I cannot understand how to address.

I need to apply explicit permission to a huge bunch of file in Onedrive for Business, breaking even the permission inheritance and the sharing inheritance. 
These files are distribuited in shared folder, so actually inherit even permission and the sharing link from the folder.

How can I address that?

Thanks!

**Comment by veronicageek** [comment]
Hi there,
ODFB is the user's personal space where only the user (and possibly an admin) should have access to it, where all content is private to the user by default. What are you trying to achieve?

**Reply by msrosellinpo** [reply]
We are migrating from Google Workspace / Google Drive to M365 / Onedrive for Business, using Mover.io to migrate that manage only permission at folder-level.

In Google Drive users could had set in the years permission at file-level, even in append and in restriction versus its parent folder.
So we are building a script that basing on the permission extracted and related to the source file apply the same permission on the target file in ODFB. 

The script works but I am able only to set permission in append versus the folder, I need to report also the permission without inheritance, breaking the inheritance first and then removing all of the associated sharing link applied on the files.
